Source/System.Management/Pash/Implementation/ExecutionVisitor.cs

```diff
@@ -212,6 +213,17 @@ object EvaluateBinaryExpression(BinaryExpressionAst binaryExpressionAst)
             }
         }
 
+        private static bool IsInstanceOfType(object leftOperand, object rightOperand)
+        {
+            var type = rightOperand as Type;
```

Comment:
The right operand is coerced to a type, actually.

```
PS Home:\> '' -is 'string'
True
```

And the right operand can be an expression:

```
'' -is [type]'string'
3 -is 'int'+32
```

Not sure right now whether that's already covered here.